src/plugins/application_config/server/opensearch_config_client.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IScopedClusterClient, Logger } from '../../../../src/core/server';
+
+import { ConfigurationClient } from './types';
+
+export class OpenSearchConfigurationClient implements ConfigurationClient {
+  private client: IScopedClusterClient;
+  private configurationIndexName: string;
+  private readonly logger: Logger;
+
+  constructor(
+    openSearchClient: IScopedClusterClient,
+    configurationIndexName: string,
+    logger: Logger
+  ) {
+    this.client = openSearchClient;
+    this.configurationIndexName = configurationIndexName;
+    this.logger = logger;
+  }
+
+  async getEntityConfig(entity: string) {
+    try {
+      const data = await this.client.asInternalUser.get({
+        index: this.configurationIndexName,
+        id: entity,
+      });
+
+      return data?.body?._source?.value || '';
+    } catch (e) {
+      const errorMessage = `Failed to get entity ${entity} due to error ${e}`;
+
+      this.logger.error(errorMessage);
+
+      throw e;
+    }
+  }
+
+  async updateEntityConfig(entity: string, newValue: string) {
+    try {
+      await this.client.asCurrentUser.index({
+        index: this.configurationIndexName,
+        id: entity,
+        body: {
+          value: newValue,
+        },
+      });
+
+      return newValue;
+    } catch (e) {
+      const errorMessage = `Failed to update entity ${entity} with newValue ${newValue} due to error ${e}`;
+
+      this.logger.error(errorMessage);
+
+      throw e;
+    }
+  }
+
+  async deleteEntityConfig(entity: string) {
+    try {
+      await this.client.asCurrentUser.delete({
+        index: this.configurationIndexName,
+        id: entity,
+      });
+
+      return entity;
+    } catch (e) {
+      if (e?.body?.error?.type === 'index_not_found_exception') {
+        this.logger.info('Attemp to delete a not found index.');
+        return entity;
+      }
+
+      if (e?.body?.result === 'not_found') {
+        this.logger.info('Attemp to delete a not found document.');
+        return entity;
```

Comment:
instead of returning entity, can we pass response with meaningful information why delete operation failed? What would be the response status code if fails?

Reply:
Addressed here. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5855/files#r1505073929 Both are happy cases per my current definition.